src/driver_client/dclient.rs

```diff
@@ -206,12 +218,14 @@ impl DriverClient {
             HBICAP_ADDR::ADDR_CPU2HIF_HBICAP_TRANSFER_SIZE,
             (binary.len() / 4) as u32,
         )?;
-        self.dma_write_by_chunks(self.cfg.ctrl_hbicap_baseaddr, DMA_RW::OFFSET, binary, 4)?;
-
+        self.dma_write(self.cfg.dma_hbicap_baseaddr, DMA_RW::OFFSET, binary)?;
         while !self.is_hbicap_ready() {
-            sleep(Duration::from_millis(1));
+            sleep(Duration::from_millis(10));
         }
         self.set_dfx_decoupling(0)?;
+        self.set_firewall_block(self.cfg.ctrl_firewall_baseaddr, false)?;
+        self.set_firewall_block(self.cfg.dma_firewall_baseaddr, false)?;
```

Comment:
I think when we want to block/unblock the firewall we always need to call both ctrl and dma.

Perhaps wrap this in a function?